python/src/nanoarrow/_lib.pyx

```diff
@@ -1177,11 +1462,435 @@ cdef class CBufferView:
+            raise RuntimeError("CArrayBuilder is already initialized")
+
+        cdef int code = ArrowArrayInitFromType(self._ptr, <ArrowType>type_id)
+        Error.raise_error_not_ok("ArrowArrayInitFromType()", code)
+
+        code = ArrowSchemaInitFromType(self.c_array._schema._ptr, <ArrowType>type_id)
+        Error.raise_error_not_ok("ArrowSchemaInitFromType()", code)
+
+        return self
+
+    def init_from_schema(self, CSchema schema):
+        if self._ptr.release != NULL:
+            raise RuntimeError("CArrayBuilder is already initialized")
+
+        cdef Error error = Error()
+        cdef int code = ArrowArrayInitFromSchema(self._ptr, schema._ptr, &error.c_error)
+        error.raise_message_not_ok("ArrowArrayInitFromType()", code)
+
+        self.c_array._schema = schema
+        return self
+
+    def start_appending(self):
+        cdef int code = ArrowArrayStartAppending(self._ptr)
+        Error.raise_error_not_ok("ArrowArrayStartAppending()", code)
+        return self
+
+    def set_offset(self, int64_t offset):
+        self.c_array._assert_valid()
+        self._ptr.offset = offset
+        return self
+
+    def set_length(self, int64_t length):
+        self.c_array._assert_valid()
+        self._ptr.length = length
+        return self
+
+    def set_null_count(self, int64_t null_count):
+        self.c_array._assert_valid()
+        self._ptr.null_count = null_count
+        return self
+
+    def resolve_null_count(self):
+        self.c_array._assert_valid()
+
+        # This doesn't apply to unions. We currently don't have a schema view
+        # or array view we can use to query the type ID, so just use the format
+        # string for now.
+        format = self.c_array.schema.format
+        if format.startswith("+us:") or format.startswith("+ud:"):
+            return self
+
+        # Don't overwrite an explicit null count
+        if self._ptr.null_count != -1:
+            return self
+
+        cdef ArrowBuffer* validity_buffer = ArrowArrayBuffer(self._ptr, 0)
+        if validity_buffer.size_bytes == 0:
+            self._ptr.null_count = 0
+            return self
+
+        # From _ArrowBytesForBits(), which is not included in nanoarrow_c.pxd
+        # because it's an internal inline function.
+        cdef int64_t bits = self._ptr.offset + self._ptr.length
+        cdef int64_t bytes_required = (bits >> 3) + ((bits & 7) != 0)
+
+        if validity_buffer.size_bytes < bytes_required:
+            raise ValueError(
+                f"Expected validity bitmap >= {bytes_required} bytes "
+                f"but got validity bitmap with {validity_buffer.size_bytes} bytes"
+            )
+
+        cdef int64_t count = ArrowBitCountSet(
+            validity_buffer.data,
+            self._ptr.offset,
+            self._ptr.length
+        )
+        self._ptr.null_count = self._ptr.length - count
+        return self
+
+    def set_buffer(self, int64_t i, CBuffer buffer, move=False):
```

Comment:
Can you add a bit of comment/docstring what this move=False vs True exactly does?

Reply:
I see below that the docstring of the public function `c_array_from_buffers` has an explanation: "Use ``True`` to move ownership of any input buffers or children to the output array.". 

I would repeat that here for when looking at the code here. And it's also not super clear to me what the consequence is exactly of doing that? Or, why would you want to set it to True?

Reply:
Thanks, that expanded docstring explanation helps a lot!